### PR TITLE
select users design

### PR DESCRIPTION
**Pull request by zanderz**

### CODE REVIEW

**Comment by zanderz** [comment]
Never got to the bolded text prompt or enter button box
image

**Comment by zanderz** [comment]
@keybase/react-hackers This is at a reasonable stopping place

File: shared/team-building/team-box.js [hunk 1]
```diff
@@ -40,28 +42,69 @@ class UserBubbleCollection extends React.PureComponent<{
...
-      onEnterKeyDown={props.onEnterKeyDown}
-      onDownArrowKeyDown={props.onDownArrowKeyDown}
-      onUpArrowKeyDown={props.onUpArrowKeyDown}
-      onBackspace={props.onBackspace}
-      searchString={props.searchString}
-    />
+  <Kb.Box2 direction="vertical" style={styles.container} fullWidth={true}>
+    <Kb.Box2 direction="horizontal" fullWidth={true}>
+      <Kb.Box2 direction="horizontal" style={styles.search}>
+        {Styles.isMobile && <Kb.Icon fontSize={22} type={'iconfont-search'} style={styles.searchIcon} />}
```
Comment (buoyad) [inline shared/team-building/team-box.js]: `{'iconfont-search'}` -> `"iconfont-search"`

File: shared/team-building/team-box.js [hunk 2]
```diff
@@ -40,28 +42,69 @@ class UserBubbleCollection extends React.PureComponent<{
...
+            props.teamSoFar.length
+              ? 'Add another username or enter to chat'
+              : 'Enter any phone number, email address, or username'
+          }
+          searchString={props.searchString}
+        />
+      </Kb.Box2>
+      {!!props.teamSoFar.length && !Styles.isMobile && <GoButton onClick={props.onFinishTeamBuilding} />}
+    </Kb.Box2>
+    <Kb.Box2 direction="horizontal" fullHeight={false} fullWidth={true} style={styles.bubbles}>
```
Comment (buoyad) [inline shared/team-building/team-box.js]: don't need `fullWidth`/`fullHeight` props if they're `false`

File: shared/team-building/user-bubble.js [hunk 3]
```diff
@@ -14,7 +15,13 @@ export type Props = {
 
 const KeybaseUserBubble = (props: Props) => (
   <Kb.Box2 className="user" direction="horizontal" style={styles.bubble}>
-    <Kb.Avatar size={bubbleSize} username={props.username} />
+    <Kb.ConnectedNameWithIcon
+      colorFollowing={true}
+      horizontal={false}
+      icon={props.service !== 'keybase' ? serviceIdToIconFont(props.service) : undefined}
```
Comment (buoyad) [inline shared/team-building/user-bubble.js]: service icons don't quite fit
image

File: shared/team-building/user-bubble.js [hunk 4]
```diff
@@ -121,17 +144,17 @@ const styles = Styles.styleSheetCreate({
...
+      height: removeSize,
+      width: removeSize,
     },
     isElectron: {
       cursor: 'pointer',
     },
   }),
 
-  removeBubbleTextAlignCenter: Styles.platformStyles({
+  removeBubbleTextAlignRight: Styles.platformStyles({
```
Comment (buoyad) [inline shared/team-building/user-bubble.js]: reason for the name change?

File: shared/team-building/user-bubble.js [hunk 5]
```diff
@@ -143,10 +166,10 @@ const styles = Styles.styleSheetCreate({
 
   removeIcon: Styles.platformStyles({
     isElectron: {
```
Comment (buoyad) [inline shared/team-building/user-bubble.js]: not centered in the bubble, `margin: 'auto'` might help
image

File: shared/common-adapters/usernames/index.js [hunk 6]
```diff
@@ -137,6 +137,7 @@ class Usernames extends React.Component<Props> {
         backgroundMode={this.props.backgroundMode}
         style={Styles.collapseStyles([containerStyle, this.props.containerStyle])}
         title={this.props.title}
+        ellipsizeMode={'tail'}
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/common-adapters/usernames/index.js]: :white_check_mark:

File: shared/common-adapters/name-with-icon/index.js [hunk 7]
```diff
@@ -86,22 +86,24 @@ class NameWithIcon extends React.Component<NameWithIconProps> {
...
         onUsernameClicked={
           this.props.clickType === 'tracker' || this.props.clickType === 'profile' ? undefined : 'profile'
         }
         type={this.props.horizontal ? 'BodySemibold' : adapterProps.titleType}
-        containerStyle={
-          this.props.horizontal ? undefined : Styles.isMobile ? undefined : styles.vUsernameContainerStyle
-        }
+        containerStyle={Styles.collapseStyles([
+          this.props.horizontal ? undefined : Styles.isMobile ? undefined : styles.vUsernameContainerStyle,
+          this.props.size === 'smaller' ? styles.smallerWidthTextContainer : {},
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/common-adapters/name-with-icon/index.js]: :white_check_mark:

File: shared/common-adapters/name-with-icon/index.js [hunk 8]
```diff
@@ -213,6 +219,15 @@ const styles = Styles.styleSheetCreate({
     maxWidth: '100%',
     width: '100%',
   },
+  smallerWidthText: {},
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/common-adapters/name-with-icon/index.js]: :white_check_mark:

File: shared/common-adapters/name-with-icon/index.js [hunk 9]
```diff
@@ -86,22 +86,24 @@ class NameWithIcon extends React.Component<NameWithIconProps> {
...
       <ConnectedUsernames
         onUsernameClicked={
           this.props.clickType === 'tracker' || this.props.clickType === 'profile' ? undefined : 'profile'
         }
         type={this.props.horizontal ? 'BodySemibold' : adapterProps.titleType}
-        containerStyle={
-          this.props.horizontal ? undefined : Styles.isMobile ? undefined : styles.vUsernameContainerStyle
-        }
+        containerStyle={Styles.collapseStyles([
+          this.props.horizontal ? undefined : Styles.isMobile ? undefined : styles.vUsernameContainerStyle,
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/common-adapters/name-with-icon/index.js]: :white_check_mark:

File: shared/team-building/team-box.js (hunk 1 above)
Comment (zanderz) [reply shared/team-building/team-box.js]: :white_check_mark:

File: shared/team-building/team-box.js (hunk 2 above)
Comment (zanderz) [reply shared/team-building/team-box.js]: :white_check_mark:

File: shared/team-building/user-bubble.js (hunk 4 above)
Comment (zanderz) [reply shared/team-building/user-bubble.js]: whoops, nvm

File: shared/team-building/user-bubble.js (hunk 5 above)
Comment (zanderz) [reply shared/team-building/user-bubble.js]: I added that and I think it's ok?

File: shared/common-adapters/name-with-icon/index.js [hunk 10]
```diff
@@ -86,22 +86,24 @@ class NameWithIcon extends React.Component<NameWithIconProps> {
       )
     }
     const usernameOrTitle = this.props.username ? (
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/common-adapters/name-with-icon/index.js]: :white_check_mark:

File: shared/common-adapters/name-with-icon/index.js [hunk 11]
```diff
@@ -227,6 +242,13 @@ const styles = Styles.styleSheetCreate({
 // Get props to pass to subcomponents (Text, Avatar, etc.)
 const getAdapterProps = (size: Size, isUser: boolean) => {
   switch (size) {
+    case 'smaller':
+      return {
+        iconSize: 48,
+        metaMargin: 6,
+        metaOneType: 'BodySmall',
+        titleType: 'BodySemibold',
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/common-adapters/name-with-icon/index.js]: :white_check_mark:

**Comment by zanderz** [comment]
Mobile didn't change much from the existing picker, so I tried to change it back the way it was.

File: shared/team-building/user-bubble.js (hunk 3 above)
Comment (zanderz) [reply shared/team-building/user-bubble.js]: @chrisnojima I wasn't able to figure out why the service icons get chopped off. I asked @buoyad and he was stumped too, can you help figure that part out?

**Comment by adamjspooner** [comment]
Nitpick: The upper corners need a border radius.
Screen Shot 2019-03-11 at 15 10 44

**Comment by adamjspooner** [comment]
More nitpicks:
* Let's drop the follow status icons.
* The close “x” is too large and off center.
Screen Shot 2019-03-11 at 15 12 48

More, not in the image attached in my last comment:
* We're missing the “x” button to dismiss the modal.
* The user rows should have more left and right padding to bring them into line with the services tab bar.

**Comment by zanderz** [comment]
@adamjspooner thanks for the review, I tried to address all those things. This should be OK except for the service icon alignment @buoyad mentioned.

**Comment by adamjspooner** [comment]
You bet!

I played around with the service icon misalignment, and a quick hack would be to relatively position the icon inside it's container. I think I used `top: 2` (or 3). Though, it doesn't solve the root of the problem. Maybe it's with the metrics when we build the font?

**Comment by zanderz** [comment]
ok I did something for that too now - not sure how right it is

File: shared/team-building/index.js [hunk 12]
```diff
@@ -168,6 +191,9 @@ const styles = Styles.styleSheetCreate({
     common: {
       paddingBottom: Styles.globalMargins.small,
     },
+    isElectron: {
+      marginLeft: Styles.globalMargins.medium,
```
Comment (buoyad) [inline shared/team-building/index.js]: shouldn't this be smaller and have a `marginRight` also to match up with the services?

Comment (zanderz) [reply shared/team-building/index.js]: ah I see, ok

**Comment by zanderz** [comment]
wow, thanks everybody, it's been a journey!

**Pull request merged**